src/libraries/System.Text.Json/source_generation/src/JsonSourceGeneratorHelper.cs

```diff
@@ -0,0 +1,499 @@
+                                {
+                                    ignoreCondition = (JsonIgnoreCondition)namedArgs[0].TypedValue.Value;
+                                }
+                            }
+                            break;
+                        case "System.Text.Json.Serialization.JsonIncludeAttribute":
+                            {
+                                hasJsonInclude = true;
+                            }
+                            break;
+                        case "System.Text.Json.Serialization.JsonNumberHandlingAttribute":
+                            {
+                                IList<CustomAttributeTypedArgument> ctorArgs = attributeData.ConstructorArguments;
+                                if (ctorArgs.Count != 1)
+                                {
+                                    throw new InvalidOperationException($"Invalid use of 'JsonNumberHandlingAttribute' detected on '{memberInfo.DeclaringType}.{memberInfo.Name}'.");
+                                }
+
+                                numberHandling = (JsonNumberHandling)ctorArgs[0].Value;
+                            }
+                            break;
+                        case "System.Text.Json.Serialization.JsonPropertyNameAttribute":
+                            {
+                                IList<CustomAttributeTypedArgument> ctorArgs = attributeData.ConstructorArguments;
+                                if (ctorArgs.Count != 1 || ctorArgs[0].ArgumentType != _stringType)
+                                {
+                                    throw new InvalidOperationException($"Invalid use of 'JsonPropertyNameAttribute' detected on '{memberInfo.DeclaringType}.{memberInfo.Name}'.");
+                                }
+
+                                jsonPropertyName = (string)ctorArgs[0].Value;
+                                // Null check here is done at runtime within JsonSerializer.
+                            }
+                            break;
+                        default:
+                            break;
+                    }
+                }
+            }
+
+            Type memberCLRType;
+            bool hasGetter;
+            bool hasSetter;
+            bool getterIsVirtual = false;
+            bool setterIsVirtual = false;
+
+            switch (memberInfo)
+            {
+                case Reflection.PropertyInfo propertyInfo:
+                    {
+                        MethodInfo setMethod = propertyInfo.SetMethod;
+
+                        memberCLRType = propertyInfo.PropertyType;
+                        hasGetter = PropertyAccessorCanBeReferenced(propertyInfo.GetMethod, hasJsonInclude);
+                        hasSetter = PropertyAccessorCanBeReferenced(setMethod, hasJsonInclude) && !setMethod.IsInitOnly();
+                        getterIsVirtual = propertyInfo.GetMethod?.IsVirtual == true;
```

Comment:
nit `== true` not necessary

Reply:
It's necessary since Get/SetMethod can be `null`.